GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1537,6 +1506,82 @@ static IEnumerable<ObjectId> TryGetParents(GitModule module, FilterInfo filterIn
             }
         }
 
+        /// <summary>
+        /// Returns the name of a file in a specific revision (following renames and merge commits)
+        /// </summary>
+        public string? GetRevisionFileName(string path, ObjectId? objectId)
+        {
+            if (FilePathByObjectId is null || objectId is null)
+            {
+                return null;
+            }
+
+            if (FilePathByObjectId.TryGetValue(objectId, out string? fileName))
```

Comment:
This unnecessary restriction can be removed:
```suggestion
            if (objectId is null)
            {
                return null;
            }

            if (FilePathByObjectId?.TryGetValue(objectId, out string? fileName) is true)
```